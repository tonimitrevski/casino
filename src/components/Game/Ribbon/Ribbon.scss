@import "../../../style/variables";

.ribbon {
  pointer-events: none;
  position: absolute;
  right: 5px;
  top: 5px;
  z-index: $z-index-game;
  overflow: hidden;
  width: 75px;
  height: 75px;
  text-align: right;

  &__label {
    font-size: 10px;
    font-weight: 700;
    color: $color-ribon-label-buttons;
    text-transform: uppercase;
    text-align: center;
    line-height: 20px;
    -webkit-transform: rotate(45deg);
    transform: rotate(45deg);
    width: 100px;
    display: block;
    background: $background-active-buttons;
    box-shadow: 0 3px 10px -5px #000;
    position: absolute;
    top: 19px;
    right: -21px;
  }
}
