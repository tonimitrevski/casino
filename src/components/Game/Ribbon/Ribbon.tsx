import React from "react";
import PropTypes from "prop-types";
import "./Ribbon.scss";

const Ribbon = (props : { type: string }) => {
  const { type } = props;
  return (
    <div className="ribbon">
      <span className="ribbon__label">{type}</span>
    </div>
  );
}

Ribbon.propTypes = {
  type: PropTypes.string,
};

export default Ribbon;
