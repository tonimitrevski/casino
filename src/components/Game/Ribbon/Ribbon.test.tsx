import React from 'react';
import {shallow} from "enzyme";
import Ribbon from "./Ribbon";

describe("Ribbon Component", () => {
  test('Test Ribbon component', () => {
    const type = "test";
    const wrapper = shallow(<Ribbon type={type} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.ribbon').length).toBe(1);
    expect(wrapper.find('.ribbon__label').length).toBe(1);
    expect(wrapper.find('.ribbon__label').text()).toBe(type);
  });
});
