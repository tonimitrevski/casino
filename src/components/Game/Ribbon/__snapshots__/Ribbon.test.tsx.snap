// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon Component Test Ribbon component 1`] = `
<div
  className="ribbon"
>
  <span
    className="ribbon__label"
  >
    test
  </span>
</div>
`;
